addons/website/static/src/components/configurator/configurator.js

```diff
@@ -0,0 +1,519 @@
+/** @odoo-module **/
+
+import rpc from 'web.rpc';
+import utils from 'web.utils';
+import weUtils from 'web_editor.utils';
+import {_t} from 'web.core';
+
+const {Component, Store, mount, QWeb} = owl;
+const {useDispatch, useStore, useGetters, useRef} = owl.hooks;
+const {Router, RouteComponent} = owl.router;
+const {whenReady} = owl.utils;
+
+const MAX_PALETTES = 16;
+
+const WEBSITE_TYPES = {
+    1: {id: 1, label: _t("a business website"), name: 'business'},
```

Comment:
Unless it's been modified since (@mart-e ?) `_t` doesn't work at the module toplevel, you need to use `_lt`.

Reply:
![Screenshot from 2021-04-02 10-31-12](https://user-images.githubusercontent.com/49784121/113398254-9478e980-939e-11eb-8483-3fb98f8840d3.png)

Just tried and it worked

Reply:
@smottet even if you have multiple languages enable and switch from one to another? Maybe this is no longer an issue?

Reply:
Just tried and yes it works correctly

Reply:
It may just happen that the combination of dependencies that you have guarantees that the translations are loaded at this point but I wouldn't trust it either way. Although now that I'm looking at it, it doesn't look like `_lt` would add any guarantees either, it delays evaluation by returning an object whose `toString` method will call `_t` when used, but this is only useful if you're sure that it will only be `toString`'ed after the translation, this is usually achieved by the fact that most JS modules define widgets or classes that will be instantiated by the web-client at some level in the class/widget hierarchy, and the web-client will wait for `session.is_bound` in its start method.

But this has me confused because you don't depend on session or on the web-client, so I'm unsure why translation works *at all*.

Reply:
@sdegueldre `_lt` will perform its lookup at each stringification, with no caching. So if you use it before loading the translations DB those will not be affected, but the uses following the loading of the translation DB will work fine. Here the labels are used in methods of components / widgets / ... so the translations should be loaded pretty much by definition.

> But this has me confused because you don't depend on session or on the web-client, so I'm unsure why translation works at all.

It's JS living in the webclient, translations should be present by definition.

Reply:
> Translations should be present pretty much by definition if you're in an odoo environment.

Except in this case we aren't, that's what I am saying: this file is standalone, the widget starts on its own on `DOMContentLoaded` using owl's `whenReady`, it does not get instantiated by the web-client or the website-root, hence there is no guarantee that the translations are loaded. It just so happens that the `website_root` is loaded before this module and that it depends on the session, which loads the translations as a side-effect, but since this page uses nothing from the website root it could probably even be culled from the assets that this page uses.

Since this widget is not instantiated by the website root, it just happens to work because the website root is created before this file is loaded and because the network latency is low and/or because the local odoo instance uses a single worker and will finish replying to the translation request before replying to the request to load this file and/or because translations are lighter than this source file.

I've looked into it with @smottet and we have converted the module-level translations to `_lt` and added an explicit wait for `session.is_bound` before mounting the component.